account_permanent_lock_move/models/account_move.py

```diff
@@ -31,11 +31,18 @@ def create(self, vals):
         result._check_lock_date()
         return result
 
+    @api.model
+    def _get_allowed_write_field(self):
+        return []
+
     @api.multi
     def write(self, vals):
         # Add _check_lock_date for write of account.move,
         # as it is not done by default
-        self._check_lock_date()
+        allowed_write_field = self._get_allowed_write_field()
+        if not set(allowed_write_field).issuperset(vals.keys()):
+            self._check_lock_date()
         result = super(AccountMove, self).write(vals)
-        self._check_lock_date()
+        if not set(allowed_write_field).issuperset(vals.keys()):
+            self._check_lock_date()
```

Comment:
This code is duplicated, maybe we could move it to a specific method which would call ```_get_allowed_write_field```.